src/relay/backend/interpreter.cc

```diff
@@ -710,17 +709,17 @@ class Interpreter : public ExprFunctor<ObjectRef(const Expr& n)>,
   Target target_;
   // Object stack.
   Stack stack_;
-  // Backend compile engine.
-  CompileEngine engine_;
+  // TE-to-TIR lowerer (compiler).
+  TECompiler compiler_;
```

Comment:
Is it possible to hold a customzied TECompiler? Say if it is NPU TECompiler, which has differnent pass flow.

Reply:
Ultimately we want to enable customized rewrites/transforms at all stages. This work is just getting us out of the go-straight-to-packed-func world so we can have a representation which makes the current te visible.

This is my first week so I'm just wrapping my head around this -- I defer to Jarad for the True Story.

Reply:
@xqdan the goal of TECompile is to have one lowering which applies initial scheduling to functions and puts them in the IRModule. After this users can customize the flow for specific targets or executors.